google/datalab/contrib/pipeline/composer/_composer.py

```diff
@@ -29,22 +30,34 @@ def __init__(self, zone, environment):
     """
     self._zone = zone
     self._environment = environment
+    self._gcs_dag_location = None
 
   def deploy(self, name, dag_string):
     client = gcs.Client()
-    bucket = client.get_bucket(self.bucket_name)
-    filename = 'dags/{0}.py'.format(name)
+    try:
+      gcs_dag_location_splits = self.gcs_dag_location.split('/')
+      bucket_name = gcs_dag_location_splits[2]
+      # Usually the splits are like ['gs:', '', 'foo_bucket', 'dags']. But we could have additional
+      # parts after the bucket. In those cases, the final file path needs to include those as well
+      additional_parts = ''
+      if len(gcs_dag_location_splits) > 4:
+        additional_parts = '/' + '/'.join(gcs_dag_location_splits[4:])
```

Comment:
Instead of reconstructing this, just use the maxsplit argument to split so that the path section won't be split in the first place.

https://docs.python.org/2/library/stdtypes.html#str.split

Reply:
Ah, cool. Done.